server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -113,6 +116,14 @@ public void setVersion(Version version) {
         this.version = version;
     }
 
+    public boolean hasFeature(final String feature) {
```

Comment:
can we have a java docs with some explanation of what the features are (or a link to where it's explained).